hal/src/rtl872x/rtl_sdk_support.cpp

```diff
@@ -221,6 +222,15 @@ extern "C" void HAL_Core_System_Reset(void) {
     }
 }
 
+extern "C" void _freertos_mfree(uint8_t *pbuf, u32 sz) {
+    if (__get_BASEPRI() != 0) {
+        // Defer freeing of memory to the SystemISRTaskQueue to be processed outside of this critical section
+        SPARK_ASSERT(system_isr_task_queue_free_memory((void*)pbuf) == 0);
+    } else {
+        vPortFree(pbuf);
```

Comment:
Use the regular `free()`, we don't normally go to heap implementation directly, only through libc malloc/free/realloc/calloc.